website/content/v1.10/kubernetes-guides/configuration/inlinemanifests.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "inlineManifests and extraManifests"
+description: "Learn what inlineManifests and extraManifests are, how they differ, and why they matter."
+---
+
+`inlineManifests` and `extraManifests` allow you to automatically apply Kubernetes resources to your cluster during initial bootstrap..
+
+Both are designed to automate the provisioning of components like CNIs and other static infrastructure, but they differ in how the manifest content is sourced and applied.
+They are not meant for deploying applications or frequently changing services.
+For those, it's better to use a GitOps or CI/CD tool.
+
+## inlineManifests
+
+`inlineManifests` are defined directly within the machine configuration file.
+The YAML content is embedded inside the `inlineManifests` section, making it ideal for tightly coupled resources that need to be provisioned as soon as the node boots up.
+
+Here’s an example of how to configure a cluster using an `inlineManifest`:
+
+```yaml
+cluster:
+  inlineManifests:
+    - name: my-app
+      contents: |
+        apiVersion: apps/v1
+        kind: Deployment
+        metadata:
+          name: my-application
+        spec:
+          # ... deployment specification
+```
+
+## extraManifests
+
+`extraManifests` are Kubernetes manifests fetched from external, unauthenticated HTTP sources such as GitHub, raw file servers, or gists.
+
+You define them in the `extraManifests` section of the machine configuration.
+They’re best suited for shared, versioned, or centrally managed resources.
+
+These manifests are pulled directly by the node during configuration.
+If the node doesn’t have network access to the HTTP endpoint hosting the manifest the installation will fail.
+
+Similarly, if the endpoint is down or returns an error, the manifest will not be applied, and the machine configuration will fail as a result.
+
+Here’s how to reference `extraManifests`:
+
+```yaml
+cluster:
+  extraManifests:
+    - "https://raw.githubusercontent.com/example/repo/main/manifest.yaml"
+    - "https://gist.githubusercontent.com/user/gist-id/raw/manifest.yaml"
+```
+
+## Resource Ordering Considerations
+
+Talos automatically sorts all manifests, including `inlineManifests`, `extraManifests`, or `built-in manifests`, before applying them in the following order:
```

Comment:
What are built-in manifests?

Reply:
they should not be in backticks, and we should probably clarify that Talos has some built-in manifests, e.g. kubelet bootstrap token, CoreDNS, etc.